### PR TITLE
feat(codegen-scala): use package imports

### PR DESCRIPTION
**Pull request by raboof**
To make code nicer to read without introducing naming conflicts for the
common case where domain and API contain the same names.

Adding this structure could hopefully also help with detecting and avoiding
conflicts later (#327).

### CODE REVIEW

File: codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala [hunk 1]
```diff
@@ -80,7 +80,8 @@ object SourceGeneratorUtils {
   def typeImport(fullyQualifiedName: FullyQualifiedName): String = {
     val name =
       if (fullyQualifiedName.parent.javaMultipleFiles) fullyQualifiedName.name
-      else fullyQualifiedName.parent.javaOuterClassname
+      else if (fullyQualifiedName.parent.javaOuterClassname.nonEmpty) fullyQualifiedName.parent.javaOuterClassname
+      else fullyQualifiedName.name
```
Comment (patriknw) [inline codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala:84]: what will happen if `java_outer_classname` is defined when generating Scala? You said that ScalaPB don't have that option so maybe things will be totally wrong if that is accidentally defined?

Ignore and fallback to right defaults for Scala or fail fast if `java_outer_classname` is defined for Scala?

Comment (raboof) [reply codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala:84]: This code is shared, so we don't "know" if we're generating Scala here.

We can make sure in the Scala `FullyQualifiedNameExtractor` that we never populate `javaOuterClassname` there, but beyond that I'm not sure there's an easy way to add more safeguards. One thing we could do is remove `PackageNaming.from(Descriptor)` and make sure this is only done in `FullyQualifiedNameExtractor`, but I haven't dared to pull that thread yet ;).

(the ideal solution IMO would be to remove `javaOuterClassname` from `FullyQualifiedName` entirely and interpret it inside the Java `FullyQualifiedNameExtractor` already, but that might not be possible/easy)

File: codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala [hunk 2]
```diff
@@ -93,43 +94,56 @@ object SourceGeneratorUtils {
   def packageAsPath(packageName: String): Path =
     Paths.get(packageName.replace(".", "/"))
 
+  class Imports(val currentPackage: String, val imports: Seq[String], val semi: Boolean) {
+    def contains(imp: String) = imports.contains(imp)
+
+    override def toString = {
+      val suffix = if (semi) ";" else ""
```
[the remark this answers is not in the document]
Comment (patriknw) [reply codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala:101]: Let's take this cleanup in a separate PR.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**